cli/tools/vendor/mod.rs

```diff
@@ -33,29 +32,33 @@ pub async fn vendor(ps: ProcState, flags: VendorFlags) -> Result<(), AnyError> {
   let output_dir = fs_util::resolve_from_cwd(&raw_output_dir)?;
   validate_output_dir(&output_dir, &flags, &ps)?;
   let graph = create_graph(&ps, &flags).await?;
-  let vendored_count =
-    build::build(&graph, &output_dir, &build::RealVendorEnvironment)?;
+  let vendored_count = build::build(
+    graph,
+    &output_dir,
+    ps.maybe_import_map.as_deref(),
+    &build::RealVendorEnvironment,
+  )?;
 
   eprintln!(
-    r#"Vendored {} {} into {} directory.
-
-To use vendored modules, specify the `--import-map` flag when invoking deno subcommands:
-  deno run -A --import-map {} {}"#,
+    concat!("Vendored {} {} into {} directory.",),
     vendored_count,
     if vendored_count == 1 {
       "module"
     } else {
       "modules"
     },
     raw_output_dir.display(),
-    raw_output_dir.join("import_map.json").display(),
-    flags
-      .specifiers
-      .iter()
-      .map(|s| s.as_str())
-      .find(|s| !is_remote_specifier_text(s))
-      .unwrap_or("main.ts"),
   );
+  if vendored_count > 0 {
+    eprintln!(
+      concat!(
+        "\nTo use vendored modules, specify the `--import-map {}` flag when ",
+        r#"invoking deno subcommands or add an `"importMap": "<path_to_vendored_import_map>"` "#,
+        "entry to your deno.json file.",
+      ),
+      raw_output_dir.join("import_map.json").display(),
```

Comment:
Why is there an actual specifier for `--import-map`, but not for `<path_to_vendored_import_map>`?

Reply:
It's more complicated and I didn't think it was worth it. It would require checking where the config is and then doing different logic for whether the config existed or didn't exist.